src/uu/whoami/src/platform/windows.rs

```diff
@@ -7,22 +7,23 @@
  * file that was distributed with this source code.
  */
 
-extern crate winapi;
+use std::ffi::OsString;
+use std::io;
+use std::os::windows::ffi::OsStringExt;
 
-use self::winapi::shared::lmcons;
-use self::winapi::shared::minwindef;
-use self::winapi::um::{winbase, winnt};
-use std::io::{Error, Result};
-use std::mem;
-use uucore::wide::FromWide;
+use winapi::shared::lmcons;
+use winapi::shared::minwindef::DWORD;
+use winapi::um::winbase;
 
-pub unsafe fn get_username() -> Result<String> {
-    #[allow(deprecated)]
-    let mut buffer: [winnt::WCHAR; lmcons::UNLEN as usize + 1] = mem::uninitialized();
-    let mut len = buffer.len() as minwindef::DWORD;
-    if winbase::GetUserNameW(buffer.as_mut_ptr(), &mut len) == 0 {
-        return Err(Error::last_os_error());
+pub fn get_username() -> io::Result<OsString> {
```

Comment:
any idea with code coverage is 0 here?

Reply:
All the tests were disabled on Windows. I'll see if I can fix it.